presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -835,6 +837,48 @@ public static Slice normalize(@SqlType("varchar(x)") Slice slice, @SqlType("varc
         return utf8Slice(Normalizer.normalize(slice.toStringUtf8(), targetForm));
     }
 
+    @Description("labels whether input strings use Unicode or Zawgyi font encoding")
+    @ScalarFunction
+    @SqlNullable
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice myanmarFontEncoding(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    {
+        if (slice == null) {
+            return null;
+        }
+        ZawgyiDetector detector = new ZawgyiDetector();
+        if (detector.getZawgyiProbability(slice.toStringUtf8()) > 0.9) {
+            return utf8Slice("zawgyi");
+        }
+        else {
+            return utf8Slice("unicode");
+        }
+    }
+
+    @Description("transforms strings using Myanmar characters to a normalized Unicode form")
+    @ScalarFunction
+    @SqlNullable
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice myanmarNormalizeUnicode(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    {
+        if (slice == null) {
+            return null;
+        }
+        ZawgyiDetector detector = new ZawgyiDetector();
+        String[] rawInputPieces = slice.toStringUtf8().split("\n");
+        String[] normalizedPieces = new String[rawInputPieces.length];
+        for (int i = 0; i < rawInputPieces.length; i++) {
+            if (detector.getZawgyiProbability(rawInputPieces[i]) > 0.9) {
+                TransliterateZ2U z2U = new TransliterateZ2U("Zawgyi to Unicode");
```

Comment:
What does the constructor input mean?

Reply:
> What does the constructor input mean?

The input to this is just used as an identifier to the `Transliterate` class used behind the scenes: 

https://github.com/google/myanmar-tools/blob/master/clients/java/src/main/java/com/google/myanmartools/TransliterateZ2U.java#L29

super class:

https://github.com/google/myanmar-tools/blob/master/clients/java/src/main/java/com/google/myanmartools/Transliterate.java#L30